src/Sylius/Bundle/CoreBundle/Fixture/Factory/ShippingMethodExampleFactory.php

```diff
@@ -55,11 +59,16 @@ public function __construct(
         RepositoryInterface $zoneRepository,
         RepositoryInterface $shippingCategoryRepository,
         RepositoryInterface $localeRepository,
-        ChannelRepositoryInterface $channelRepository
+        ChannelRepositoryInterface $channelRepository,
+        ?RepositoryInterface $taxCategoryRepository = null
     ) {
         $this->shippingMethodFactory = $shippingMethodFactory;
         $this->zoneRepository = $zoneRepository;
         $this->shippingCategoryRepository = $shippingCategoryRepository;
+        $this->taxCategoryRepository = $taxCategoryRepository;
+        if($this->taxCategoryRepository === null) {
+            trigger_error('Not passing a taxCategory repo is deprcated and will be removed');
```

Comment:
```suggestion
            @trigger_error(sprintf('Not passing a $taxCategoryRepository to %s constructor is deprecated since Sylius 1.4 and will be removed in Sylius 2.0.', self::class), \E_USER_DEPRECATED);
```